contracts/GovStorage.sol

```diff
@@ -50,26 +61,470 @@ contract GovStorage is IGovernance {
+            _proposal.endTime,
+            _proposal.proposer,
+            _proposal.status,
+            _proposal.approvalMode,
+            _proposal.targets,
+            _proposal.values,
+            _proposal.calldatas,
+            _proposal.descriptionHash
+        );
+    }
+
+    function getNumberOfVotingDays(
+        uint128 _class
+    ) public view returns(uint256) {
+        return votingReward[_class].numberOfVotingDays;
+    }
+
+    function getTotalVoteTokenPerDay(
+        uint128 _class,
+        uint128 _nonce,
+        uint256 _votingDay
+    ) public view returns(uint256) {
+        return totalVoteTokenPerDay[_class][_nonce][_votingDay];
+    }
+
+    function increaseTotalVoteTokenPerDay(
+        uint128 _class,
+        uint128 _nonce,
+        uint256 _votingDay,
+        uint256 _amountVoteTokens
+    ) public {
+        totalVoteTokenPerDay[_class][_nonce][_votingDay] += _amountVoteTokens;
+    }
+
+    function getNumberOfDBITDistributedPerDay(
+        uint128 _class
+    ) public view returns(uint256) {
+        return votingReward[_class].numberOfDBITDistributedPerDay;
+    }
+ 
+    function setProposal(
+        uint128 _class,
+        uint128 _nonce,
+        uint256 _startTime,
+        uint256 _endTime,
+        address _proposer,
+        ProposalApproval _approvalMode,
+        address[] memory _targets,
+        uint256[] memory _values,
+        bytes[] memory _calldatas,
+        string memory _description
+    ) public {
+        proposal[_class][_nonce].startTime = _startTime;
+        proposal[_class][_nonce].endTime = _endTime;
+        proposal[_class][_nonce].proposer = _proposer;
+        proposal[_class][_nonce].approvalMode = _approvalMode;
+        proposal[_class][_nonce].targets = _targets;
+        proposal[_class][_nonce].values = _values;
+        proposal[_class][_nonce].calldatas = _calldatas;
+        proposal[_class][_nonce].descriptionHash = keccak256(bytes(_description));
+    }
+
+    function setProposalStatus(
+        uint128 _class,
+        uint128 _nonce,
+        ProposalStatus _status
+    ) public {
+        proposal[_class][_nonce].status = _status;
+    }
+
+    /**
+    * @dev set a proposal class info for a given class and index
+    * @param _class proposal class
+    * @param _index index in the proposal class info array
+    * @param _value the new ven value of the proposal class info
+    */
+    function setProposalClassInfo(
+        uint128 _class,
+        uint256 _index,
+        uint256 _value
+    ) public {
+        proposalClassInfo[_class][_index] = _value;
+    }
+
+    function getProposalNonce(
+        uint128 _class
+    ) public view returns(uint128) {
+        return proposalNonce[_class];
+    }
+
+    function setProposalNonce(
+        uint128 _class,
+        uint128 _nonce
+    ) public {
+        proposalNonce[_class] = _nonce;
+    }
+
+    //==== FROM EXECUTABLE
+
+    function updateGovernanceContract(
```

Comment:
this should have the `modifier`condition for checking that current governance contract is only being modified by the debondOperator (or via the proposal).

Reply:
Yes, I will add the modifier. It was there in the previous version. I removed them for testing purposes. I forgot to put them back